rero_ils/modules/documents/utils_mef.py

```diff
@@ -0,0 +1,234 @@
+                self.increment_count(self.count_found, ref)
+                # delete mef $schema
+                metadata.pop('$schema', None)
+                # create local contribution
+                return Contribution.create(data=metadata, dbcommit=True,
+                                           reindex=True)
+            else:
+                # online contribution has no IdREf, GND or RERO
+                self.increment_count(self.count_no_data, ref)
+        except Exception as err:
+            # no online contribution found
+            self.increment_count(self.count_no_mef, ref)
+
+    def process(self):
+        """Process replacement."""
+        if self.verbose:
+            click.echo(f'Found: {self.query.count()}')
+        for hit in self.query.source('pid').scan():
+            doc = Document.get_record_by_id(hit.meta.id)
+            new_contributions = []
+            changed = False
+            for contribution in doc.get('contribution', []):
+                new_contributions.append(contribution)
+                ref_type = contribution['agent'].get(
+                    'identifiedBy', {}).get('type', '').lower()
+                if ref_type in self.cont_types + ['rero']:
+                    ref_pid = contribution['agent'].get(
+                        'identifiedBy', {}).get('value')
+                    cont = self.get_local(ref_type=ref_type, ref_pid=ref_pid)
+                    if not cont:
+                        cont = self.get_online(ref_type=ref_type,
+                                               ref_pid=ref_pid)
+                    if cont:
+                        # change the contribution to linked contribution
+                        for cont_type in self.cont_types:
+                            if cont.get('cont_type'):
+                                changed = True
+                                url = f'{self.mef_url}/{cont_type}/' \
+                                    f'{cont[cont_type]["pid"]}'
+                                new_contributions[-1]['agent'] = {
+                                    '$ref': url,
+                                    'type': contribution['agent']['type']
+                                }
+                                break
+            if changed:
+                doc['contribution'] = new_contributions
+                doc.update(data=doc, dbcommit=True, reindex=True)
+        return self.counts
+
+
+class ReplaceMefIdentifiedBySubjects(ReplaceMefIdentifiedByContribution):
+    """Replace MEF identified by subjects class."""
+
+    def __init__(self, run=False, verbose=False):
+        """Constructor."""
+        self.query = DocumentsSearch() \
+            .filter('exists', field='subjects.identifiedBy') \
+            .filter('bool', should=[
+                Q('term', subjects__type='bf:Person'),
+                Q('term', subjects__type='bf:Organisation')
+            ])
+        super().__init__(run=run, verbose=verbose)
```

Comment:
according to previous review, just can just override the `def _query_filter()` method
```python
def _query_filter():  # could be override by subclasses
     return DocumentsSearch() \
            .filter('exists', field='subjects.identifiedBy') \
            .filter('bool', should=[
                Q('term', subjects__type='bf:Person'),
                Q('term', subjects__type='bf:Organisation')
            ])
```